nautobot/dcim/views.py

```diff
@@ -4044,50 +3990,34 @@ def post(self, request, pk):
             },
         )
 
-
-class VirtualChassisRemoveMemberView(ObjectPermissionRequiredMixin, GetReturnURLMixin, View):
-    queryset = Device.objects.all()
-
-    def get_required_permission(self):
-        return "dcim.change_device"
-
-    def get(self, request, pk):
-        device = get_object_or_404(self.queryset, pk=pk, virtual_chassis__isnull=False)
-        form = ConfirmationForm(initial=request.GET)
-
-        return render(
-            request,
-            "dcim/virtualchassis_remove_member.html",
-            {
-                "device": device,
-                "form": form,
-                "return_url": self.get_return_url(request, device),
-            },
+    @action(detail=True, methods=["get", "post"], url_path="remove-member", url_name="remove_member")
+    def remove_member(self, request, pk=None):
+        device = get_object_or_404(
+            Device.objects.restrict(request.user, "change"), pk=pk, virtual_chassis__isnull=False
```

Comment:
Hm, note that this view/URL is expecting a *device* PK, not a *virtual-chassis* PK. Rolling this into the viewset **might** not be the correct thing to do. (See also my earlier comment about changing the URL pattern for this view as well...)